### PR TITLE
perf: optimize string builtins with utf8 byte compare

### PR DESCRIPTION
**Pull request by arirubinstein**
### What code changed, and why?

Optimizes `strings.contains`, `strings.endswith`, and `strings.startswith` builtins to use direct UTF-8 byte comparison (`memcmp`) instead of Swift's default String methods. 

This matches Go's `strings.Contains`/`HasPrefix`/`HasSuffix` semantics - raw byte comparison with no Unicode normalization - and improves performance by avoiding Foundation overhead. 

Added a comprehensive test suite covering multi-byte UTF-8 (emoji, CJK), Unicode normalization edge cases, error handling, and boundary conditions.

### CODE REVIEW

File: Sources/Rego/Builtins/Strings.swift [hunk 1]
```diff
@@ -97,7 +147,30 @@ extension BuiltinFuncs {
...
+            return .boolean(true)
+        }
+        let baseCount = base.utf8.count
+        let searchCount = search.utf8.count
+        if baseCount > searchCount {
+            return .boolean(false)
+        }
+
+        // Fast path: contiguous memcmp
+        if let result = search.utf8.withContiguousStorageIfAvailable({ sBuf in
```
Comment (philipaconrad) [inline Sources/Rego/Builtins/Strings.swift:161]: [comment]: I remember seeing this `Array<T>` [method](https://developer.apple.com/documentation/swift/array/withcontiguousstorageifavailable(_:)) when I was last browsing around the Apple Developer docs, and it's nice to see a live use case for it! 🙂

File: Sources/Rego/Builtins/Strings.swift [hunk 2]
```diff
@@ -97,7 +147,30 @@ extension BuiltinFuncs {
...
+            base.utf8.withContiguousStorageIfAvailable({ bBuf in
+                guard let sBase = sBuf.baseAddress, let bBase = bBuf.baseAddress else {
+                    return false
+                }
+                return memcmp(sBase + (sBuf.count - bBuf.count), bBase, bBuf.count) == 0
```
Comment (philipaconrad) [inline Sources/Rego/Builtins/Strings.swift:166]: [comment]: I had to stare at this line for long time to understand why the `search` string is getting the pointer bump. The reason is because (IMO) we chose horrifically bad parameter names, and I was thinking "search" was the suffix to search for. 🤦 

Names like "search" and "suffix" might have been vastly better choices for the parameter names, but that's an upstream OPA problem, and no fault of your code. I think it should be correct as written here.

This is also the part of the code I was most worried about and thus didn't want to review hastily. Apologies about the delay on reviewing this bit!

**Review by philipaconrad — APPROVED** [review]
Thank you @arirubinstein for putting this PR together!

It took me a hot second to go dig up the docs for some of the parts I was not familiar with, but I have confidence now that this PR should not break anything, and does exactly what's intended. It should give us a healthy speed boost on string operations too! 😄 

Review commentary below, no changes or revisions needed:

**Comment by philipaconrad** [comment]
ℹ️ Rebasing the PR through the Github Web UI (to catch up with `main`) before merging...

**Comment by philipaconrad** [comment]
I'm not usually a fan of `users.noreply.github.com` emails being used in signoffs-- it's literally "click a button on github to become unreachable"-- but given that OPA has merged over 200 commits with emails of that form in the signoff, I guess the precedent has already been set. 🤷‍♂️ 


The CNCF doesn't overly much seem to care either, based on what I've been able to [dig up](https://github.com/cncf/foundation/issues/383), so I think we should be good for merge.
